### ISSUE TITLE
When deploying the application in Docker (locally, on cloud Linux, or in Render), both the worker and server containers fail to connect to Redis, resulting in an `ECONNREFUSED` error.

### ISSUE THREAD

**Issue by eliasbiondo** [twentyhq/twenty#7768]
## Bug Description

When deploying the application in Docker (locally, on cloud Linux, or in Render), both the worker and server containers fail to connect to Redis, resulting in an `ECONNREFUSED` error. The application logs show the following error:

[Nest] 7  - 10/16/2024, 10:27:38 PM   ERROR [ExceptionHandler] connect ECONNREFUSED 127.0.0.1:6379
Error: connect ECONNREFUSED 127.0.0.1:6379
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1495:16)

## Technical inputs

- The error suggests that the application is trying to connect to Redis on `127.0.0.1:6379`, which might be incorrect in the context of Docker containers.

**Comment by Hitarthsheth07** [comment]
This is because your redis is not running properly. 

Follow this instructions: https://twenty.com/developers/local-setup#step-4:-set-up-a-redis-database-(cache)

Then try:
`redis-cli`

if you get an error message you're not connected to Redis server.
You can try restarting the redis server by the following command: 

`/etc/init.d/redis-server restart`
OR 

If you don't get error, type PING in cli. If Redis is up and running it will return PONG. 

Also, join discord for asking questions: https://discord.gg/txbw7Uxe

**Comment by rosscdh-tpg** [comment]
in docker latest
seems to be a hardcoded 127.0.0.1:6379 in the codebase as its not referring to REDIS_URL

**Comment by rosscdh** [comment]
I rebuilt docker image from source... same issue

![Screenshot 2024-10-17 at 18 56 31](https://github.com/user-attachments/assets/e0cbc986-fdb3-4ecc-badb-3d596aa27ce6)
![Screenshot 2024-10-17 at 18 56 22](https://github.com/user-attachments/assets/a212d156-c14c-4f3c-9a46-715cf9809c1f)

**Comment by FelixMalfait** [comment]
Probably linked to https://github.com/twentyhq/twenty/pull/7736/files but I can't find what's wrong 🤔

**Comment by rosscdh** [comment]
yep.. im looking too... it seems to ignore the ENV_VAR and i see no ref to 127.0.0.1 which imlpies perpas that the interface to the underlying library is wrong... but looks ok to me

**Comment by rosscdh** [comment]
ive commented REDIS_URL out in the compose and in the .env and get the error its missing.. so... something.. WEIRD is going on.. maybe the underlying redis library doesnt like "redis://redis" which would be kinda....

**Comment by rosscdh** [comment]
nope.. changed REDIS_URL=redis://redis.server:6379 and renamed redis service to the name.. .no joy..


```
twenty-server-1                | Error: connect ECONNREFUSED 127.0.0.1:6379
twenty-server-1                |     at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1495:16) {
twenty-server-1                |   errno: -111,
twenty-server-1                |   code: 'ECONNREFUSED',
twenty-server-1                |   syscall: 'connect',
twenty-server-1                |   address: '127.0.0.1',
twenty-server-1                |   port: 6379
twenty-server-1                | }
```

**Comment by rosscdh** [comment]
@FelixMalfait  perhaps https://github.com/redis/node-redis/issues/2416 similar in the redis package you use?

**Comment by FelixMalfait** [comment]
Aaaah I think I understood why! 
You're pulling the 0.31 docker image which still uses the old env variable.
But if you use 1-click install or read the doc, it only talks about REDIS_URL which would only work on the unstable docker image tagged with "latest"

**Comment by stnby** [comment]
Having same issue on fresh deployment. My theory is that this was merged without making a new release https://github.com/twentyhq/twenty/pull/7736
And broke deployment script.

Temporary fix is to add this to both server and worker containers:
REDIS_HOST: redis

**Comment by FelixMalfait** [comment]
@stnby same comment at the same time haha :)

**Comment by FelixMalfait** [comment]
We should update https://github.com/twentyhq/twenty/blob/main/install.sh#L48 to use the tagged release instead of the "main" branch. And we should probably update the docs on step 2 to replace `curl -O https://raw.githubusercontent.com/twentyhq/twenty/main/packages/twenty-docker/docker-compose.yml` by something ` https://github.com/twentyhq/twenty/tree/v{x.y.z}` indicating {x.y.z} should be replaced by the version number

**Comment by rosscdh** [comment]
```
TAG=v0.31.2

# POSTGRES_ADMIN_PASSWORD=replace_me_with_a_strong_password

PG_DATABASE_HOST=db:5432
REDIS_URL=redis://redis:6379/0
REDIS_HOST=redis
```

seems to work!

Good work I appreciate the input and well done, on what looks to be a fantastic CRM. I look forward to putting it through its paces

**Comment by Raphael-Quinones** [comment]
I'm curious as to why when I rebuilt the docker image from source, it doesn't work. But when pulling from the latest it does work when you apply all the fixes in this thread. Is the one on github not the same with the image on dockerhub?

**Comment by FelixMalfait** [comment]
Re-opening as I think we should solve the root cause in the 1-click install script (using main branch instead of version branch).

(Also fyi we realized REDIS_URL wasn't taken into account properly, a fix is being pushed today #7956)

**Comment by FelixMalfait** [comment]
OK we can close I created this as a followup: #7968
